### PR TITLE
update with managed readonly policy and metrics inline policies

### PR DESCRIPTION
**Pull request by ole3021**
* Add AWS mangaed ReadonlyAccess policy to avoid size limite issue.
* New Metric related Role and policy
* Add parameter for stackreporter and Serverless AWS account reference in different stage.

### CODE REVIEW

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 1]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+                  Ref: AWS::AccountId
+      Path: "/"
+  #
+  ##
+  ###
+  #### SERVERLESS ROLE READ PERMISSIONS
+  ###
+  ##
+  #
+  PolicyOfServerlessRoleReadGroupAtoD:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]:
```suggestion
  ServerlessReadOnlyPolicyAtoD
```

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 2]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+              # https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ListStreams.html
+              - "dynamodb:ListStreams"
+              # Grants permission to return an array of table names associated with the current account and endpoint.
+              # https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ListTables.html
+              - "dynamodb:ListTables"
+              # Grants permission to list all tags on an Amazon DynamoDB resource.
+              # https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ListTagsOfResource.html
+              - "dynamodb:ListTagsOfResource"
+            Resource: "*"
+  PolicyOfServerlessRoleReadGroupEtoK:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]:
```suggestion
  ServerlessReadOnlyPolicyEtoK
```

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 3]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+              # https://docs.aws.amazon.com/kinesis/latest/APIReference/API_ListStreamConsumers.html
+              - "kinesis:ListStreamConsumers"
+              # Grants permission to list your streams.
+              # https://docs.aws.amazon.com/kinesis/latest/APIReference/API_ListStreams.html
+              - "kinesis:ListStreams"
+              # Grants permission to list the tags for the specified Amazon Kinesis stream.
+              # https://docs.aws.amazon.com/kinesis/latest/APIReference/API_ListTagsForStream.html
+              - "kinesis:ListTagsForStream"
+            Resource: "*"
+  PolicyOfServerlessRoleReadGroupLtoX:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]:
```suggestion
  ServerlessReadOnlyPolicLtoZ
```

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 4]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+              - "xray:ListTagsForResource"
+            Resource: "*"
+  #
+  ##
+  ###
+  #### SERVERLESS ROLE WRITE PERMISSIONS
+  ###
+  ##
+  #
+  PolicyOfServerlessRoleWriteGroup:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]:
```suggestion
  ServerlessWritePolicy
```

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 5]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+              - "s3:PutBucketNotification"
+            Resource: "*"
+  #
+  ##
+  ###
+  #### SERVERLESS ROLE Metrics PERMISSIONS
+  ###
+  ##
+  #
+  PolicyOfServerlessRoleMetricsGroup:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]:
```suggestion
ServerlessMetricsStreamPolicy
```

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 6]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+              # The bucket name will include unique id to avoid bucket duplication
+              - "arn:aws:s3:::serverless.metrics-firehose-backup-*"
+  #
+  ##
+  ###
+  #### SERVERLESS ROLE ASSUME PERMISSIONS
+  ###
+  ##
+  #
+  PolicyOfServerlessRoleAssumeGroup:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]:
```suggestion
  ServerlessAssumePolicy:
```

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 7]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/ServerlessMetricsRole"
+  #
+  ##
+  ###
+  #### SERVERLESS Metrics ROLE PERMISSIONS
+  ###
+  ##
+  #
+  PolicyOfServerlessMetricsRoleStreamGroup:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]:
```suggestion
  ServerlessMetricsStreamPolicy:
```

File: instrumentation/aws/iam-role-cfn-template.yaml [hunk 8]
```diff
@@ -24,608 +24,1689 @@ Resources:
...
+
+              # Grants permission to write a single data record into an Amazon Kinesis Firehose delivery stream.
+              # https://docs.aws.amazon.com/firehose/latest/APIReference/API_PutRecord.html
+              - "firehose:PutRecord"
+              # Grants permission to write multiple data records into a delivery stream in a single call, which can achieve higher throughput per producer than when writing single records.
+              # https://docs.aws.amazon.com/firehose/latest/APIReference/API_PutRecordBatch.html
+              - "firehose:PutRecordBatch"
+            Resource:
+              - !Sub "arn:aws:firehose:*:${AWS::AccountId}:deliverystream/serverless_metrics-firehose"
+  PolicyOfServerlessMetricsRoleFirehoseGroup:
```
Comment (Mmarzex) [inline instrumentation/aws/iam-role-cfn-template.yaml]: Why do we need this policy?

**Review by Mmarzex — CHANGES_REQUESTED** [review]

File: instrumentation/aws/iam-role-cfn-template.yaml (hunk 8 above)
Comment (ole3021) [reply instrumentation/aws/iam-role-cfn-template.yaml]: This is the permissions that needed for the firehose deployed in users AWS account.  This is also the permissions auto added when manually create a firehose through AWS UI.

**Comment by ole3021** [comment]
@Mmarzex  Has changed the policy names. (Replace the Metrics with Monitoring, make it consistent with our doc.).

**Pull request closed without merging**